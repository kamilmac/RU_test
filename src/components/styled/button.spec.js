import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

it('should not throw an error when called', () => {
    <Button />
})

it('renders button', () => {
    const wrapper = shallow(<Button
        fontSize={20}
        color="black"
        bgColor="white"
        height={20}
        width={20}
    />)
    expect(wrapper).toMatchSnapshot()
})

it('Should have the correct styled(component) displayName', () => {
    expect(Button.displayName).toBe('styled.button')
})
