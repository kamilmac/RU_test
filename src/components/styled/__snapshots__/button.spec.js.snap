exports[`test renders button 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="kfaaFG"
    color="black"
    fontSize={20}
    height={20}
    width={20} />,
  "nodes": Array [
    <button
      className="kfaaFG"
      color="black"
      fontSize={20}
      height={20}
      width={20} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <styled.button
        bgColor="white"
        color="black"
        fontSize={20}
        height={20}
        width={20} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "bgColor": "white",
          "color": "black",
          "fontSize": 20,
          "height": 20,
          "width": 20,
        },
        "refs": Object {},
        "state": Object {
          "generatedClassName": "kfaaFG",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="kfaaFG"
          color="black"
          fontSize={20}
          height={20}
          width={20} />,
        "_debugID": 2,
        "_renderedOutput": <button
          className="kfaaFG"
          color="black"
          fontSize={20}
          height={20}
          width={20} />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <styled.button
    bgColor="white"
    color="black"
    fontSize={20}
    height={20}
    width={20} />,
}
`;
